clang/include/clang/Basic/Diagnostic.h

```diff
@@ -1333,13 +1263,7 @@ class DiagnosticBuilder : public StreamingDiagnostic {
 public:
   /// Copy constructor.  When copied, this "takes" the diagnostic info from the
   /// input and neuters it.
-  DiagnosticBuilder(const DiagnosticBuilder &D) : StreamingDiagnostic() {
-    DiagObj = D.DiagObj;
-    DiagStorage = D.DiagStorage;
```

Comment:
Well well, `DiagStorage` might be allocated via `DiagStorageAllocator` so this looked like a potential memory manager mess. Now, I fixed this (see cpp).